### PR TITLE
Kyverno 2021 annual review

### PR DESCRIPTION
**Pull request by JimBugwadia**
annual review for Kyverno (https://kyverno.io/)

### CODE REVIEW

File: reviews/2021-Kyverno-annual.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+We are using the following metrics as key indicators of community health:
+* [Stargazers](https://kyverno.devstats.cncf.io/d/81/community-health?orgId=1&var-repo_name=All&var-metric=Stargazers&var-table=swatchers&var-pref=all&var-met1=watch&var-met2=watch&from=now-1y&to=now): 331% annual growth from 524 -> 1735.
+* [Issue Creators](https://kyverno.devstats.cncf.io/d/81/community-health?orgId=1&var-repo_name=All&var-metric=Issue%20creators&var-table=scommunity_health&var-pref=&var-met1=chealthissue&var-met2=&from=now-1y&to=now): 500% annual growth from 45 -> 225.
+* [Code Committers](https://kyverno.devstats.cncf.io/d/81/community-health?orgId=1&var-repo_name=All&var-metric=Code%20committers&var-table=scommunity_health&var-pref=&var-met1=chealthcommit&var-met2=&from=now-1y&to=now): 500% annual growth from 18 -> 90.
+* [Code Commenters](https://kyverno.devstats.cncf.io/d/81/community-health?orgId=1&var-repo_name=All&var-metric=Code%20commenters&var-table=scommunity_health&var-pref=&var-met1=chealthcomment&var-met2=&from=now-1y&to=now): 509% annual growth from 22 -> 112.
+
+## Maintainers
+
+Kyverno currently has 5 maintainers from 4 different companies:
```
Comment (dims) [inline reviews/2021-Kyverno-annual.md:49]: Can you please update this @JimBugwadia ? Looks like we are upto 8 maintainers now.

https://github.com/kyverno/kyverno/blob/main/MAINTAINERS.md#maintainers

**Comment by dims** [comment]
Questions:
- Is there a plan for [Kyverno Certification](https://learn.nirmata.com/explore) program mentioned in https://kyverno.io/community/ going to be community governed at some point?
- I see End User and Contributors meeting in https://kyverno.io/community/ - are these recorded? (if so missing link to play list)

File: reviews/2021-Kyverno-annual.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+* **Community and Awareness**: Kyverno has grown from initially having maintainers from a single company, to maintainers from multiple organizations. The community has also grown significantly as reflected by [DevStats](https://kyverno.devstats.cncf.io). While Kyverno awareness has grown, more work is needed to make the community fully aware of all of its features and capabilities.
+
+> What are the current goals of the project? For example, are you working on major new features? Or are you concentrating on adoption or documentation?
+
+1. **Scalability**: Kyverno is increasingly being used in large production clusters. We aim to dramatically improve scalability in the subsequent releases.
+
+2. **Security**: In addition to current security processes, a goal for Kyverno is to improve software supply chain security and achieve [SLSA Level 4](https://slsa.dev/levels).
+
+3. **Adoption**: We aim to continue growing Kyverno adoption. A survey conducted by Nirmata at KubeCon US 2021 showed Kyverno adoption is currently around 8%. Our goal is to exceed 30% adoption by Q4 2022.
```
Comment (dims) [inline reviews/2021-Kyverno-annual.md:106]: is there a plan for this that is public? (what steps/when etc)

File: reviews/2021-Kyverno-annual.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+A list of companies providing commercial services and solutions for Kyverno is available [here](https://kyverno.io/support/).
+
+## Project Goals
+
+> How has the project performed against its goals since the last review? (We won't penalize you if your goals changed for good reasons.)
+
+The main goals of the project have been in the following categories:
+
+* **Complex policy support**: Several new features such as [JMESPath support](https://kyverno.io/docs/writing-policies/variables/), [API Server Lookups](https://kyverno.io/docs/writing-policies/external-data-sources/#variables-from-kubernetes-api-server-calls), and [foreach](https://kyverno.io/docs/writing-policies/validate/#foreach) have been introduced to allow declarative processing of complex policy logic.
```
[the remark this answers is not in the document]
Comment (JimBugwadia) [reply reviews/2021-Kyverno-annual.md:95]: Yes. The designs were mostly discussed via GDocs and slack. 

We recently created a repo for design proposals and added links to recently implemented and pending designs: 

   https://github.com/kyverno/KDP

File: reviews/2021-Kyverno-annual.md (hunk 2 above)
Comment (JimBugwadia) [reply reviews/2021-Kyverno-annual.md:111]: Here is the tracking issue for the [SLSA Level 3](https://github.com/kyverno/kyverno/issues/3119) work which is planned as an LFX Mentorship program that just started.

**Comment by JimBugwadia** [comment]
> Questions:
> 
> * Is there a plan for [Kyverno Certification](https://learn.nirmata.com/explore) program mentioned in https://kyverno.io/community/ going to be community governed at some point?

Not at the moment, but this is a good idea! There have been discussions of updating the certification with hands-on components, so makes sense to drive that via an community governance process. 

> * I see End User and Contributors meeting in https://kyverno.io/community/ - are these recorded? (if so missing link to play list)

The meetings used to be streamed, but stopped due to zoom security settings. The recordings are available at:

https://docs.google.com/document/d/1kFd4fpAoHS56mRHr73AZp9wknk1Ehy_hTB_KA7gJuy0/edit#

https://docs.google.com/document/d/10Hu1qTip1KShi8Lf_v9C5UVQtp7vz_WL3WVxltTvdAc/edit#

File: reviews/2021-Kyverno-annual.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+> How can the CNCF help you achieve your upcoming goals?
+
+Kyverno has benefited greatly from the CNCF and its community, especially in the areas of awareness and adoption. The Kyverno project has also participated in the [LFX Mentorship program](https://mentorship.lfx.linuxfoundation.org/#projects_all), KubeCon project office hours, and the Community Infrastructure Lab.
+
+Some areas where we can use help are:
+ 
+* Best practices for automated scale testing.
+* A security assessment and guidance on fuzzing and other security best practices to improve the Kyverno security posture. 
+* Continued support for webinar, blogs, and other community activities to increase visibility and adoption.
+* Add Kyverno to the Certified Kubernetes Security Specialist (CKS) examination and any other CNCF materials that reference the Open Policy Agent (OPA) project, to help drive awareness and promote fairness across projects.
```
[the remark this answers is not in the document]
Comment (JimBugwadia) [reply reviews/2021-Kyverno-annual.md:124]: Here is one recent example in the docs (which was fixed):

   https://github.com/kubernetes/website/pull/31879/files

For CKS, this question has come up in the community from folks who have taken the exams in various forums:

https://twitter.com/eduardoriveror/status/1480575568722673676?s=20

Here are some references to OPA in CKS training materials:

- https://jonathan18186.medium.com/certified-kubernetes-security-specialist-cks-preparation-part-6-open-policy-agent-opa-6d01501bb0d7
- https://github.com/kodekloudhub/certified-kubernetes-security-specialist-cks-course/blob/main/docs/05-Minimize-Microservice-Vulnerabilities/10-Open-Policy-Agent-(OPA).md
- https://bswen.com/2021/05/cks-prepare-cks-with-me-11.html

**Comment by amye** [comment]
I could have sworn we got you your own Zoom account, I've opened a ticket in Servicedesk to track work for that.

**Pull request merged**